bundles/remote_services/topology_manager/src/topology_manager.c

```diff
@@ -212,12 +187,12 @@ celix_status_t topologyManager_rsaAdded(void * handle, service_reference_pt unus
 	remote_service_admin_service_t *rsa = (remote_service_admin_service_t *) service;
 	celix_logHelper_log(manager->loghelper, CELIX_LOG_LEVEL_INFO, "TOPOLOGY_MANAGER: Added RSA");
 
-	celixThreadMutex_lock(&manager->rsaListLock);
-	arrayList_add(manager->rsaList, rsa);
-    celixThreadMutex_unlock(&manager->rsaListLock);
 
-	// add already imported services to new rsa
-    celixThreadMutex_lock(&manager->importedServicesLock);
+    celixThreadMutex_lock(&manager->lock);
+
+    arrayList_add(manager->rsaList, rsa);
```

Comment:
nitpick: please replace with celix_ prefix array list api:
`celix_arrayList_add(manager->rsaList, rsa);`

etc for the rest of the arrayList without a celix_ prefix usage.